### PR TITLE
PEOPLE-105 rewrite teams specs

### PR DESCRIPTION
**Pull request by mczuchnowski**
**Changes:**
- remove specs for unused view (role show spec)
- small changes i scheduling spec
- total refactor of teams spec (using traits + SitePrism)

**Jira:**
https://netguru.atlassian.net/browse/PEOPLE-105

### CODE REVIEW

File: spec/features/teams_spec.rb [hunk 1]
```diff
@@ -1,75 +1,32 @@
...
+  let!(:archived_user) { create(:user, first_name: 'Archived Arthur', archived: true) }
   let!(:no_role_user) { create(:user, first_name: 'Norole Nicola') }
-
-  let!(:hidden_user) do
-    create(
-      :user,
-      first_name: 'Hidden Amanda', primary_role: hidden_role, teams: [team]
-    )
-  end
+  let!(:hidden_user) { create(:user, first_name: 'Hidden Amanda', primary_role: hidden_role,
```
Comment (netguru-hound) [inline spec/features/teams_spec.rb]: Avoid using `{...}` for multi-line blocks.<br>Block body expression is on the same line as the block start.

File: spec/features/teams_spec.rb [hunk 2]
```diff
@@ -1,75 +1,32 @@
...
-  let!(:non_dev_user) do
-    create(:user, first_name: 'Nondev Nigel', primary_role: non_dev_role)
-  end
-  let!(:non_dev_position) { create(:position, :primary, user: non_dev_user, role: non_dev_role) }
-
-  let!(:archived_user) do
-    create(:user, first_name: 'Archived Arthur', archived: true)
-  end
-
+  let!(:dev_user) { create(:user, :admin, :developer, first_name: added_user_first_name,
```
Comment (netguru-hound) [inline spec/features/teams_spec.rb]: Avoid using `{...}` for multi-line blocks.<br>Block body expression is on the same line as the block start.

File: spec/features/teams_spec.rb [hunk 3]
```diff
@@ -1,75 +1,32 @@
...
   let!(:no_role_user) { create(:user, first_name: 'Norole Nicola') }
-
-  let!(:hidden_user) do
-    create(
-      :user,
-      first_name: 'Hidden Amanda', primary_role: hidden_role, teams: [team]
-    )
-  end
+  let!(:hidden_user) { create(:user, first_name: 'Hidden Amanda', primary_role: hidden_role,
+    teams: [team]) }
```
Comment (netguru-hound) [inline spec/features/teams_spec.rb]: Align the elements of a hash literal if they span more than one line.<br>Expression at 16, 20 should be on its own line.

File: spec/features/teams_spec.rb [hunk 4]
```diff
@@ -1,75 +1,32 @@
...
-    create(:user, first_name: 'Nondev Nigel', primary_role: non_dev_role)
-  end
-  let!(:non_dev_position) { create(:position, :primary, user: non_dev_user, role: non_dev_role) }
-
-  let!(:archived_user) do
-    create(:user, first_name: 'Archived Arthur', archived: true)
-  end
-
+  let!(:dev_user) { create(:user, :admin, :developer, first_name: added_user_first_name,
+      last_name: added_user_last_name) }
```
Comment (netguru-hound) [inline spec/features/teams_spec.rb]: Align the elements of a hash literal if they span more than one line.<br>Expression at 11, 40 should be on its own line.

File: spec/features/teams_spec.rb (hunk 2 above)
Comment (mczuchnowski) [inline spec/features/teams_spec.rb]: This makes it much more readable, saves lines and stops your "line is too long" comments.

File: spec/features/teams_spec.rb (hunk 3 above)
Comment (mczuchnowski) [inline spec/features/teams_spec.rb]: Same as above

File: spec/features/teams_spec.rb (hunk 4 above)
Comment (mczuchnowski) [inline spec/features/teams_spec.rb]: See above

File: spec/features/teams_spec.rb (hunk 1 above)
Comment (mczuchnowski) [inline spec/features/teams_spec.rb]: See comment below

**Comment by mejcz** [comment]
LGTM :+1:

**Pull request merged**